data_structures/k_tree/k_tree.py

```diff
@@ -0,0 +1,79 @@
+class Node:
+    """defines each node used to build the k-ary tree"""
+    def __init__(self, val, input_list=[]):
+        self.val = val
+        self.children = []
+        for child in input_list:
+            self.children.append(Node(child))
+
+    def __repr__(self):
+        # return 'node children: {} '.format(lambda x: for x in self.children: return x.val)
+        return str(self.val) 
+
+    def __str__(self):
+        return 'node value: {}'.format(self.val)
+
+
+class Ktree:
+    """defines k-ary tree"""
+    def __init__(self, val, input_list=[]):
```

Comment:
A K-tree does not need a root when it is created so `val` should probably be optional.